plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeHistoryTable.java

```diff
@@ -0,0 +1,195 @@
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        if (commitInfoEntries.isEmpty()) {
+            return new FixedPageSource(ImmutableList.of());
+        }
+        return new FixedPageSource(buildPages(session));
+    }
+
+    private List<Page> buildPages(ConnectorSession session)
+    {
+        PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
+        TimeZoneKey timeZoneKey = session.getTimeZoneKey();
+
+        commitInfoEntries.forEach(commitInfoEntry -> {
+            pagesBuilder.beginRow();
+
+            pagesBuilder.appendBigint(commitInfoEntry.getVersion());
+            pagesBuilder.appendTimestampTzMillis(commitInfoEntry.getTimestamp(), timeZoneKey);
+            write(commitInfoEntry.getUserId(), pagesBuilder);
+            write(commitInfoEntry.getUserName(), pagesBuilder);
+            write(commitInfoEntry.getOperation(), pagesBuilder);
+            if (checkNonNull(commitInfoEntry.getOperationParameters(), pagesBuilder)) {
+                pagesBuilder.appendVarcharVarcharMap(commitInfoEntry.getOperationParameters());
+            }
+            if (checkNonNull(commitInfoEntry.getJob(), pagesBuilder)) {
+                writeCommitInfoEntryJobBlock(commitInfoEntry.getJob(), pagesBuilder.nextColumn());
+            }
+            if (checkNonNull(commitInfoEntry.getNotebook(), pagesBuilder)) {
+                writeCommitInfoEntryNotebookBlock(commitInfoEntry.getNotebook(), pagesBuilder.nextColumn());
+            }
+            write(commitInfoEntry.getClusterId(), pagesBuilder);
+            pagesBuilder.appendBigint(commitInfoEntry.getReadVersion());
+            write(commitInfoEntry.getIsolationLevel(), pagesBuilder);
+            write(commitInfoEntry.isBlindAppend(), pagesBuilder);
+
+            pagesBuilder.endRow();
+        });
+
+        return pagesBuilder.build();
+    }
+
+    private void writeCommitInfoEntryJobBlock(CommitInfoEntry.Job job, BlockBuilder blockBuilder)
+    {
+        BlockBuilder rowBuilder = blockBuilder.beginBlockEntry();
+
+        write(job.getJobId(), rowBuilder);
+        write(job.getJobName(), rowBuilder);
+        write(job.getRunId(), rowBuilder);
+        write(job.getJobOwnerId(), rowBuilder);
+        write(job.getTriggerType(), rowBuilder);
+
+        blockBuilder.closeEntry();
+    }
+
+    private void writeCommitInfoEntryNotebookBlock(CommitInfoEntry.Notebook notebook, BlockBuilder blockBuilder)
+    {
+        BlockBuilder rowBuilder = blockBuilder.beginBlockEntry();
+        write(notebook.getNotebookId(), rowBuilder);
+        blockBuilder.closeEntry();
+    }
+
+    private static boolean checkNonNull(Object object, PageListBuilder pagesBuilder)
+    {
+        if (object == null) {
+            pagesBuilder.appendNull();
```

Comment:
this is strange that this method actually appends something it is very counterintuitive

Reply:
Removing the method and using an explicit null check instead.